### ISSUE TITLE
Appium unable to see accessibility id on UIButton on UITableVIew's HeaderView for section

### ISSUE THREAD

**Issue by milosandrejic93** [appium/appium#20613]
### Do I have the most recent component updates?

- [X] I use the most recent available driver/plugin and server versions

### Is the component officially supported by the Appium team?

- [X] I have verified the component repository is present under the Appium organization in GitHub

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Appium is unable to see accessibility id on a UIButton, which is on a HeaderView for a section in UITableView. On the HeaderView there are 3 UIViews - 1 UIImageVIew, 1 UILabel and one UIButton, the parent view for all 3 views is the HeaderView. 

Appium cannot see this, because when I open Appium Inspector, I can see that in view hierarchy there are actually two UIButtons(this is not the case when you look in XCode, everything is fine) and because there are 2 UIButtons with the same accessibility id, it cannot be seen by Appium.

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/9c337192-f470-435a-b01f-1985555aa989">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/7d36ff66-1d27-4076-81a0-06e4d5a198e7">

<img width="1512" alt="image" src="https://github.com/user-attachments/assets/57ed3d4c-55d4-4191-93da-ffa74a9908a7">

### Expected Behavior

Appium can see  accessibility id set on UIButton which is normally placed on a HeaderView for a UITableView's section.

### Minimal Reproducible Example

I uploaded a simple project with this situation.

[TableViewFooterButtonDemo.zip](https://github.com/user-attachments/files/17133847/TableViewFooterButtonDemo.zip)

### Environment

- Operating system: MacOS 14.2 (23C64)
- Appium server version (output of `appium --version`): 2.11.4
- Appium driver(s) and their version(s) (`appium driver list`): - xcuitest@7.27.0 [installed (npm)]
- Appium plugin(s) and their version(s) (`appium plugin list`): - appium-reporter-plugin@1.1.0-beta.06 [installed (npm)]
- Node.js version (output of `node --version`): v22.9.0
- `npm` version (output of `npm --version`): 10.8.3
- Real device or emulator/simulator: Simulator iPhone 15 Pro(iOS 17.5)(also same behavior on all other simulators)

**Comment by KazuCocoa** [comment]
> this is not the case when you look in XCode, everything is fine

Appium gets elements via XCTest framework, which is different from Xcode.

What could you see about the element via https://appium.github.io/appium-xcuitest-driver/latest/reference/execute-methods/#mobile-source command with `format: description`?

**Comment by IgorP-Grip** [comment]
[mobile_source_description.txt](https://github.com/user-attachments/files/17160597/mobile_source_description.txt)

Thank you for your response. Please find attached txt file with command output. It is listed there twice also, button identifier is 'MAGIC_BUTTON!' 

Couple of things to consider:
- if view contains only one subview everything looks fine, problem persists only if there is 2 or more subviews in uitableview header/footer view
- element is visible and tests are passing 

let us know if we can provide more info, or if you have any idea how to bypass this issue.

**Comment by mykola-mokhnach** [comment]
> Thank you for your response. Please find attached txt file with command output. It is listed there twice also, button identifier is 'MAGIC_BUTTON!'

If it's listed twice there as well then this is how XCTest (and the Appium's XCUITest driver uses exactly the same) parses application accessibility. Try to either create an issue to Apple or ask at their forums why a single button creates two accessibility instances.

**Comment by KazuCocoa** [comment]
Yea, it is how XCTest by Apple parsed the view. Appium itself depends on it.

**Comment by mykola-mokhnach** [comment]
Closed as third party issue
